helper-cli/src/main/kotlin/commands/DownloadResultsFromPostgresCommand.kt

```diff
@@ -0,0 +1,189 @@
+            " upload-result-to-postgres command."
+) {
+    private val outputDir by option(
+        "--output-dir", "-o",
+        help = "The directory to write the ORT results to."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = false, canBeDir = true, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val tableName by option(
+        "--table-name",
+        help = "The name of the table which holds the ORT results."
+    ).required()
+
+    private val columnName by option(
+        "--column-name",
+        help = "The name of the JSONB column containing the ORT result."
+    ).required()
+
+    private val configFile by option(
+        "--config",
+        help = "The path to the ORT configuration file that configures the scan results storage."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .default(ortConfigDirectory.resolve(ORT_CONFIG_FILENAME))
+
+    private val configArguments by option(
+        "-P",
+        help = "Override a key-value pair in the configuration file. For example: " +
+                "-P ort.scanner.storages.postgres.schema=testSchema"
+    ).associate()
+
+    private val startId by option(
+        "--start-id",
+        help = "The smallest row-id to download."
+    ).convert { it.toInt() }
+        .default(-1)
+
+    override fun run() {
+        if (!outputDir.exists()) outputDir.mkdirs()
+
+        val localStorage = OrtResultStorage(outputDir)
+        val localRowIds = localStorage.getIds()
+
+        openDatabaseConnection().use { connection ->
+            val remoteRowIds = fetchRowIds(connection)
+            val missingRowIds = remoteRowIds - localRowIds
+            println("Requested the download of ${remoteRowIds.size} of which ${missingRowIds.size} are missing.")
+
+            missingRowIds.forEachIndexed { i, rowId ->
+                println("[${i + 1} / ${missingRowIds.size}] downloading row $rowId")
+
+                val duration = measureTime {
+                    val ortResultJson = fetchOrtResult(connection, rowId)
+
+                    if (ortResultJson != null) localStorage.add(rowId, ortResultJson)
+                }
+
+                println("[${i + 1} / ${missingRowIds.size}] took ${duration.toString(DurationUnit.SECONDS)}s.")
+            }
+        }
+    }
+
+    private fun fetchOrtResult(connection: Connection, rowId: Int): String? {
+        // TODO: The transmission is relatively slow currently due to lack of compression. Check whether
+        // the gzip extension can help speed this up https://github.com/pramsey/pgsql-gzip.
+
+        val query = "SELECT t.$columnName::TEXT AS result FROM $tableName t WHERE id = ?;"
```

Comment:
If you would reuse the `OrtResults` table from the upload command you could get rid of the SQL related boilerplate code.

Reply:
I'd keep this for now, in particular as I hope to make some PG specific optimizations, see comment above.